float32[] x
float32[] y
float32[] theta
uint8 TYPE_NOTMOVING=0
uint8 TYPE_ROTATE=1
uint8 TYPE_FORWARD=2
uint8 TYPE_BACKWARD=3
uint8 type
